DLM_code/Src/dlm-manage_data_aquisition.c

```diff
@@ -112,8 +112,24 @@ void set_bucket_size(U8 sending_dam, void* UNUSED,
         above_bucket_node->next = bucket_node;
     }
 
-    // set the number of parameters in this bucket
+    // set the number of parameters in this bucket. No parameters have been added yet
+    bucket_node->bucket.params_added = 0;
     bucket_node->bucket.num_of_params = num_of_params;
+
+    // free any memory that may have been used for the old parameter array
+    free(bucket_node->bucket.param_ids);
+
+    // malloc some memory for the new parameter array
+    bucket_node->bucket.param_ids = (BUCKET_PARAM_INFO*)malloc(num_of_params * sizeof(BUCKET_PARAM_INFO));
```

Comment:
This malloc call and the check for a null pointer could into a single conditional, but I'd say it's personal preference on style. This way is pretty clear ngl